File: app/services/export_service.py
def export_csv(data: list) -> str:
    return "\n".join(",".join(str(v) for v in row) for row in data)


def export_json(data: list) -> str:
    import json

    return json.dumps(data)


def export_xml(data: list) -> str:
    return "<data>" + "".join(f"<item>{r}</item>" for r in data) + "</data>"


def run_export(data: list, fmt: str) -> str:
    import sys

    handler = getattr(sys.modules[__name__], f"export_{fmt}", None)
    if handler is None:
        raise ValueError(f"Unknown export format: {fmt}")
    return handler(data)

File: web/src/services/exportService.ts
import * as self from "./exportService";

export function exportCsv(data: any[]): string {
  return data.map((row) => Object.values(row).join(",")).join("\n");
}

export function exportJson(data: any[]): string {
  return JSON.stringify(data);
}

export function exportXml(data: any[]): string {
  return "<data>" + data.map((r) => `<item>${r}</item>`).join("") + "</data>";
}

export function runExport(data: any[], fmt: string): string {
  const fnName = `export${fmt.charAt(0).toUpperCase()}${fmt.slice(1)}`;
  const handler = (self as any)[fnName];
  if (!handler) {
    throw new Error(`Unknown export format: ${fmt}`);
  }
  return handler(data);
}
